test/serializers/attribute_test.rb

```diff
@@ -137,6 +137,33 @@ def false
         end
       end
 
+      def test_multiple_conditional_attributes_on_one_key
```

Comment:
and maybe to reduce complexity, use booleans for if? instead of a variable? idk. Might be too nit-picky. You already found the bug. lol